CMakeLists.txt

```
@@ -383,6 +375,16 @@ else()
   set(VTK_FOUND OFF)
 endif()
 
+#VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND if the version we require is not found.
+option(WITH_QT "Build QT Front-End" TRUE)
+if(WITH_QT)
+  find_package(Qt5 5.9.5 COMPONENTS Concurrent OpenGL Widgets)
+  if(Qt5_FOUND)
+    message(STATUS "Qt5 version: ${Qt5_VERSION}")
+  else()
+    message(STATUS "Qt5 is not found.")
+  endif()
+endif()
```

Got another idea related to `QT_DISABLE_DEPRECATED_BEFORE`: Shouldn't it also work to change this code here to:
```suggestion
option(WITH_QT "Build QT Front-End" TRUE)
if(WITH_QT)
  find_package(Qt5 5.9.5 COMPONENTS Concurrent OpenGL Widgets)
  if(Qt5_FOUND)
    message(STATUS "Qt5 version: ${Qt5_VERSION}")
    target_compile_definitions(Qt5::Core PUBLIC QT_DISABLE_DEPRECATED_BEFORE=0x050900)
  else()
    message(STATUS "Qt5 is not found.")
  endif()
endif()
```

On this way we need to define `0x050900` only once and each target which is using Qt getting this macro automatically (therefore now `PUBLIC`). Can you check the compiler settings if I'm correct?

---

Yes and no - it depends. As long as `Qt5::Core` is passed as a public dependency (default), it will affect downstream projects. It will stop only when a target that uses `Qt5::Core` is not added publicly (so it is transitive).

Do we have any library which depends on Qt and is intended to be used by other projects?

If yes, this is indeed no option, but then we could add directly after `find_package(Qt5 ...)` `set(QT_DISABLE_DEPRECATED_BEFORE=0x050900)` and use this variable later in the `target_compile_definitions` call

---

Hm, we don't have anything in PCLConfig.cmake.in that finds Qt, so we probably don't have any projects that is used down-stream.
And as long you just use your own CMakeLists and find_package(pcl/qt) it shouldn't be any problem.

---

Hm, didn't work:
```
CMake Error at CMakeLists.txt:384 (target_compile_definitions):
  target_compile_definitions may only set INTERFACE properties on IMPORTED
  targets
```

---

Could you try [this](https://stackoverflow.com/a/55081943)?

---

Couldn't get it work either. Nor if I only set the INTERFACE options. They didn't get appended to the projects.

---

ok, thanks for trying it :-)

---

Found this in our code (may be relevant for #4619)

```
	# Original official CMake workaround for Qt5 issue [https://gitlab.kitware.com/cmake/cmake/issues/16915] was taken and adapted:
	get_property(core_options TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_OPTIONS)
	string(REPLACE "-fPIC" "" new_core_options ${core_options})
	set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_OPTIONS ${new_core_options})
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
	set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler=-fPIC")
```

So it seems possible to modify a property of `Qt5::Core`

---

Hmm, it does seem to work. Its just added to additional options, instead of preprocessor definitions. Think this is fine?

---

Had to use `INTERFACE_COMPILE_DEFINITIONS` 👍